atc/worker/artifact_source.go

```diff
@@ -149,12 +164,33 @@ func (source *artifactSource) StreamTo(
 		err = source.p2pStreamTo(ctx, destination)
 	}
 
+	if err != nil {
+		return err
+	}
+
 	// Inc counter if no error occurred.
-	if err == nil {
-		metric.Metrics.VolumesStreamed.Inc()
+	metric.Metrics.VolumesStreamed.Inc()
+
+	usedResourceCache, found, err := source.resourceCacheFactory.FindResourceCacheByID(source.volume.GetResourceCacheID())
+	if err != nil {
+		logger.Error("artifactSource-StreamTo-failed-to-find-resource-cache", err)
+		return nil
```

Comment:
Is there a reason why we are returning nil instead of the err here? Is it because you are thinking that it is okay for this method to pass even if we don't initialize the resource cache? If that is your reason, I'm also wondering the same thing but the only thing that makes me feel like it should return an error is that this is not supposed to happen. Like there is something that is going wrong with the database if this was to happen so maybe it would make sense to just fail early rather than covering up the error.

Reply:
This piece of code is trying to mark a streamed volume as a resource cache if source volume is a resource cache. If an error occurs, we just log the error and discard the error, so that it fall back to normal process.